src/Symfony/Bundle/SecurityBundle/Resources/config/security.xml

```diff
@@ -195,5 +197,12 @@
             <argument type="service" id="security.token_storage" />
             <argument type="service" id="security.encoder_factory" />
         </service>
+
+        <!-- Cache Warmers -->
+        <service id="security.cache_warmer.expression_cache_warmer" class="Symfony\Bundle\SecurityBundle\CacheWarmer\ExpressionCacheWarmer">
+            <tag name="kernel.cache_warmer"></tag>
+            <argument type="tagged" tag="security.expression"></argument>
```

Comment:
instead of relying on a tag, which creates an external extension point, you could inject this in the DI extension (based on `$this->expressions`)

Reply:
True :+1:

Reply:
Is this what you meant?

https://github.com/symfony/symfony/pull/26660/commits/adab5cec0ec07598c3f66ced07857226a5c58404